steam-client/src/main/java/com/avenga/steamclient/steam/asyncclient/steamuser/callback/MarketingMessageCallback.java

```diff
@@ -0,0 +1,67 @@
+package com.avenga.steamclient.steam.asyncclient.steamuser.callback;
+
+import com.avenga.steamclient.enums.EMarketingMessageFlags;
+import com.avenga.steamclient.generated.MsgClientMarketingMessageUpdate2;
+import com.avenga.steamclient.model.GlobalID;
+import com.avenga.steamclient.steam.asyncclient.callbackmanager.BaseCallbackMessage;
+import com.avenga.steamclient.util.stream.BinaryReader;
+import com.avenga.steamclient.util.stream.MemoryStream;
+import lombok.Getter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+
+/**
+ * This callback is fired when the client receives a marketing message update.
+ */
+@Getter
+public class MarketingMessageCallback extends BaseCallbackMessage {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MarketingMessageCallback.class);
+
+    private Date updateTime;
+    private Collection<Message> messages;
+
+    public MarketingMessageCallback(MsgClientMarketingMessageUpdate2 body, byte[] payload) {
+        updateTime = new Date(body.getMarketingMessageUpdateTime() * 1000L);
+
+        List<Message> msgList = new ArrayList<>();
+
+        try (BinaryReader binaryReader = new BinaryReader(new MemoryStream(payload))) {
+            for (int i = 0; i < body.getCount(); i++) {
```

Comment:
emm. should it really be good-old for loop? forEach mb?

Reply:
We know only amount of the messages stored in payload byte array and we don't have list of the elements to iterate.